### PR TITLE
Client authentication

### PR DESCRIPTION
**Pull request by tochman**
PT: https://www.pivotaltracker.com/story/show/178943570

* adds api keys to encrypted credentials
* adds authentication method to the application controller
* restricts service and search controllers

All requests to the API need to include a header:

```
API_KEY: "77f17e3dea2ddda1fabbbe99fef37efb3ed228ab"
```

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,7 +10,8 @@ gem 'puma', '~> 5.0'
 gem 'rack-cors', require: 'rack/cors'
 gem 'rails', '~> 6.1.3', '>= 6.1.3.2'
 gem 'slack-notify'
-gem 'devise_token_auth'
+# gem 'devise_token_auth'
```
Comment (aonomike) [inline Gemfile:13]: do we need to delete this?

Comment (aonomike) [inline Gemfile:13]: do we need to delete this?

**Review by aonomike — APPROVED** [review]
LGTM

File: Gemfile (hunk 1 above)
Comment (tochman) [reply Gemfile]: No, I'd like to keep this comment around for now.

**Review by SevaDer14 — APPROVED** [review]

**Review by 1CIM — APPROVED** [review]
👍

**Pull request merged**